File: Back/Api/Api/Models/Product.cs
using MongoDB.Bson.Serialization.Attributes;


namespace Api
{
    public class Product
    {
        [BsonId]
        [BsonElement("_id")]
        public int _id { get; set; }

        [BsonElement("name")]
        public string name { get; set; }

        [BsonElement("category")]
        public string category { get; set; }

        [BsonElement("professorPrice")]
        public double professorPrice { get; set; }

        [BsonElement("studentPrice")]
        public double studentPrice { get; set; }

        [BsonElement("weight")]
        public int weight { get; set; }
        
        [BsonElement("description")]
        public string description { get; set; }


    }
}

File: Web/src/app/classes/product.ts
export class Product {
    id: number;
    name: String;
    category: String;
    professorPrice: number;
    studentPrice: number;
    weight: number;
    description: String;
    

    constructor(id: number, name: String, category: String, description: String, professorPrice: number, studentPrice: number, weight: number){
        this.id = id;
        this.name = name;
        this.category = category;
        this.description = description;
        this.professorPrice = professorPrice;
        this.studentPrice = studentPrice;
        this.weight  = weight;
    }


}

export class SellProduct{
    product: Product;
    cantity: number;
    
    constructor(product: Product) {
        this.product = product;
        this.cantity = 1;
    }

}

export class BuyProduct {
    _id : number;
    idProduct : number;
    code : number;
    date : string;
    quantity : number;

    constructor(_id : number, idProduct : number, code : number, date : string, quantity : number){
        this._id = _id;
        this.idProduct = idProduct;
        this.code = code;
        this.date = date;
        this.quantity = quantity;
    }

}


// let Product1 = new Product(1, "dsak", "dska", "dksadksa", 12, 12,12);